### PR TITLE
MULE-11378: Allow SDK operations and sources to return List<Message>

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/MetadataOutputDelegate.java [hunk 1]
```diff
@@ -144,4 +149,34 @@
...
   }
+
+  private MetadataType adaptToListIfNecessary(MetadataType resolvedType, Object key, MetadataContext metadataContext)
+      throws MetadataResolvingException {
+    if (!(component.getOutput().getType() instanceof ArrayType)) {
+      return resolvedType;
+    }
+
+    MetadataType outputType = ((ArrayType) component.getOutput().getType()).getType();
+    String typeId = getTypeId(outputType).orElse(null);
```
Comment (alegmarra) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/MetadataOutputDelegate.java]: `org.mule.runtime.extension.api.util.ExtensionMetadataTypeUtils#getId` already handles the optional and fallbacks to the classname if missing but of JAVA format (safely)

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/TypesFactory.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+   * @param outputType     the type of the payload
+   * @param attributesType the type of the attributes
+   * @return an {@link ObjectType}
+   */
+  public static ObjectType buildMessageType(BaseTypeBuilder typeBuilder,
+                                            MetadataType outputType,
+                                            MetadataType attributesType) {
+
+    ObjectTypeBuilder messageType = typeBuilder.objectType().id(Message.class.getName());
+    messageType.addField().key("payload").required(true).value(outputType);
```
Comment (alegmarra) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/TypesFactory.java]: constants in Message?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/MetadataOutputDelegate.java (hunk 1 above)
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/MetadataOutputDelegate.java]: done

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/TypesFactory.java (hunk 2 above)
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/TypesFactory.java]: not in the message, but in the TypesFactory

**Pull request merged**